fdb-record-layer-core/src/test/resources/NestedMessage.json

```diff
@@ -0,0 +1,57 @@
+{
+  "records": {
+    "dependency": [
+      "record_metadata_options.proto"
+    ],
+    "message_type": [
+      {
+        "name": "RecordTypeUnion",
+        "field": [
+          {
+            "name": "_T2",
+            "number": 1,
+            "type": "TYPE_MESSAGE",
+            "type_name": "T2"
+          }
+        ]
+      },
+      {
+        "name": "T1",
+        "field": [
+          {
+            "name": "ID",
+            "number": 1,
+            "type": "TYPE_INT64"
+          }
+        ]
+      },
+      {
+        "name": "T2",
+        "field": [
+          {
+            "name": "ID",
+            "number": 1,
+            "type": "TYPE_INT64"
+          },
+          {
+            "name": "T1",
+            "number": 2,
+            "type": "TYPE_MESSAGE",
+            "type_name": "T1"
```

Comment:
Do we have tests here that attempt to do things like use different package specifications for the type name here? I think we have tests for that in the unit test, but maybe not through these JSON files. That level of testing may be fine